lib/OpenQA/Git.pm

```diff
@@ -21,45 +22,68 @@ sub config ($self, $args = undef) {
     return $app->config->{'scm git'};
 }
 
-sub _prepare_git_command ($self, $args = undef) {
-    my $dir = $args->{dir} // $self->dir;
-    if ($dir !~ /^\//) {
-        my $absolute_path = abs_path($dir);
-        $dir = $absolute_path if ($absolute_path);
+sub _validate_attributes ($self) {
+    for my $mandatory_property (qw(app dir user)) {
+        die "no $mandatory_property specified" unless $self->$mandatory_property();
     }
-    return ('git', '-C', $dir);
 }
 
-sub _format_git_error ($result, $error_message) {
-    if ($result->{stderr} or $result->{stdout}) {
-        $error_message .= ': ' . $result->{stdout} . $result->{stderr};
+sub _run_cmd ($self, $args, $options = {}) {
+    my $expand_git_path = $options->{expand_git_path} // 1;
+    my $use_ssh = $options->{use_ssh} // 0;
+
+    my @cmd;
+
+    if ($use_ssh) {
+        push @cmd, 'env', 'GIT_SSH_COMMAND="ssh -oBatchMode=yes"';
     }
-    return $error_message;
+
+    push @cmd, $self->_prepare_git_command($expand_git_path), @$args;
+
+    my $result = run_cmd_with_log_return_error(\@cmd);
+    if (!$result->{status}) {
+        $self->app->log->error("Git command failed: @cmd - Error: $result->{stderr}");
+    }
+    return $result;
 }
 
-sub _validate_attributes ($self) {
-    for my $mandatory_property (qw(app dir user)) {
-        die "no $mandatory_property specified" unless $self->$mandatory_property();
+sub _prepare_git_command ($self, $expand_git_path) {
```

Comment:
I think expand is really the wrong word here, sorry

Reply:
changed to `include_git_path`